src/SelfTimedReadBackOutputNode.cpp

```diff
@@ -0,0 +1,209 @@
+        "}\n"};
+    auto fragmentString = QString{
+        "#version 150\n"
+        "uniform sampler2D iFrame;\n"
+        "in vec2 uv;\n"
+        "out vec4 fragColor;\n"
+        "void main() {\n"
+        "    fragColor = texture(iFrame, uv);\n"
+        "}\n"};
+
+    auto shader = QSharedPointer<QOpenGLShaderProgram>(new QOpenGLShaderProgram());
+
+    if (!shader->addShaderFromSourceCode(QOpenGLShader::Vertex, vertexString)) {
+        emit fatal("Could not compile vertex shader");
+        return nullptr;
+    }
+    if (!shader->addShaderFromSourceCode(QOpenGLShader::Fragment, fragmentString)) {
+        emit fatal("Could not compile fragment shader");
+        return nullptr;
+    }
+    if (!shader->link()) {
+        emit fatal("Could not link shader program");
+        return nullptr;
+    }
+
+    return shader;
+}
+
+void STRBONOpenGLWorker::initialize(QSize size) {
+    makeCurrent();
+    m_shader = loadBlitShader();
+    if (m_shader.isNull()) return;
+
+    auto fmt = QOpenGLFramebufferObjectFormat{};
+    fmt.setInternalTextureFormat(GL_RGBA);
+    m_fbo = QSharedPointer<QOpenGLFramebufferObject>::create(size, fmt);
+
+    m_size = size;
+    m_pixelBuffer.resize(4 * size.width() * size.height());
+
+    m_timer = new QTimer(this);
+    connect(m_timer, &QTimer::timeout, this, &STRBONOpenGLWorker::onTimeout);
+
+    emit initialized();
+}
+
+void STRBONOpenGLWorker::setInterval(long msec) {
+    if (m_timer == nullptr) {
+        qWarning() << "Node not ready, ignoring setInterval";
+        return;
+    }
+    m_timer->setInterval(msec);
+}
+
+void STRBONOpenGLWorker::start() {
+    if (m_timer == nullptr) {
+        qWarning() << "Node not ready, ignoring start";
+        return;
+    }
+    m_timer->start();
+}
+
+void STRBONOpenGLWorker::stop() {
+    if (m_timer == nullptr) {
+        qWarning() << "Node not ready, ignoring start";
+        return;
+    }
+    m_timer->start();
+}
+
+void STRBONOpenGLWorker::onTimeout() {
+    auto d = m_p.toStrongRef();
+    if (d.isNull()) return; // SelfTimedReadBackOutputNode was deleted
+    SelfTimedReadBackOutputNode p(d);
+
+    makeCurrent();
+    GLuint texture = p.render();
+
+    if (texture == 0) {
+        qWarning() << "No frame available";
+        return;
+    }
+
+    auto vao = p.chain().vao();
+
+    glClearColor(0, 0, 0, 0);
+    glDisable(GL_DEPTH_TEST);
+    glDisable(GL_BLEND);
+
+    m_fbo->bind();
+    // VAO??
+    glViewport(0, 0, m_size.width(), m_size.height());
+
+    m_shader->bind();
+    glActiveTexture(GL_TEXTURE0);
+    glBindTexture(GL_TEXTURE_2D, texture);
+    glTexParameteri(GL_TEXTURE_2D,GL_TEXTURE_MAG_FILTER,GL_LINEAR);
+    glTexParameteri(GL_TEXTURE_2D,GL_TEXTURE_MIN_FILTER,GL_LINEAR);
+
+    m_shader->setUniformValue("iFrame", 0);
+
+    vao->bind();
+    glDrawArrays(GL_TRIANGLE_STRIP, 0, 4);
+    vao->release();
+
+    glReadPixels(0, 0, m_size.width(), m_size.height(), GL_RGBA, GL_UNSIGNED_BYTE, m_pixelBuffer.data());
```

Comment:
And this is fine because it's from an FBO, not a texture?

Reply:
Yeah, `ReadPixels` is much nicer about the returned data not having padding and stuff